src_c/_sdl2/video.pyx

```diff
@@ -940,3 +966,57 @@ cdef class Renderer:
         if not dest:
             return self.get_viewport()
         return dest
+
+    def draw_line(self, p1, p2):
+        # https://wiki.libsdl.org/SDL_RenderDrawLine
+        res = SDL_RenderDrawLine(self._renderer,
+                                 p1[0], p1[1],
+                                 p2[0], p2[1])
+        if res < 0:
+            raise error()
+
+    def draw_point(self, point):
+        # https://wiki.libsdl.org/SDL_RenderDrawPoint
+        res = SDL_RenderDrawPoint(self._renderer,
+                                  point[0], point[1])
+        if res < 0:
+            raise error()
+
+    def draw_rect(self, rect):
+        # https://wiki.libsdl.org/SDL_RenderDrawRect
+        cdef SDL_Rect _rect = pgRect_AsRect(rect)
+        res = SDL_RenderDrawRect(self._renderer, &_rect)
+        if res < 0:
+            raise error()
+
+    def fill_rect(self, rect):
+        # https://wiki.libsdl.org/SDL_RenderFillRect
+        cdef SDL_Rect _rect = pgRect_AsRect(rect)
+        res = SDL_RenderFillRect(self._renderer, &_rect)
+
+        if res < 0:
+            raise error()
+
+    def get_surface(self, depth=32):
```

Comment:
- Maybe get_surface/read_surface should be to_surface() instead? We have from_surface in Texture.

- For the use case of saving the Renderer to disk/network every frame... Having it take a destsurface would be pretty fast, because it could allow you to reuse a surface and not malloc it every frame.

Reply:
I prefer either read_surface() or to_surface() to get_surface(), because the latter might make it sound like it's just returning a stored surface if you don't know how it works

Reply:
`destsurface=None`?

Reply:
I prefer read_surface() than get_surface() since it sounds like you can get the **surface reference** of the texture, like how `pygame.display.get_surface()` does.

`def read_surface(dest=None)`,`def read_surface(surf=None)`or`def read_surface(surface=None)`?

Reply:
I think those drawing functions can be merged into pygame.draw or pygame.gfxdraw, too.

Reply:
Good point about the get_surface() perhaps seeming like it would return a reference. We also have pygame.image.tostring, frombuffer, fromstring.

Reply:
I guess to_surface is a bit more consistent with Texture.from_surface

Reply:
Then `to_surface()` seems like the best at least moment.